### PR TITLE
feat: add `tableSnapshot` helper

### PR DESCRIPTION
**Pull request by blambillotte**
Generates a human-readable string representation of a table for easy matching using `toMatchInlineSnapshot`
```ts
    expect(tableSnapshot(r)).toMatchInlineSnapshot(`
      "
      | Name                     | Value |
      | ------------------------ | ----- |
      | Row 1                    | 200   |
      | Row 2 with a longer name | 300   |
      | Row 3                    | 1000  |
      "
    `);
```

**Example from lot sequence sheet:**
```ts
      expect(tableSnapshot(r)).toMatchInlineSnapshot(`
        "
        |     |          |                  |          | Plot information  | Exterior options | Floor plan options | Add-ons          | Interior options   |               |
        |     | Address  | Construction RTS | Delivery | Plan              | Spec level       | Sellable SF        | Net Buildable SF | Gross Buildable SF | Permitable SF | Bedrooms | Full baths | Half baths | Elevation   | Exterior scheme |                     |      |      |  |
        | --- | -------- | ---------------- | -------- | ----------------- | ---------------- | ------------------ | ---------------- | ------------------ | ------------- | -------- | ---------- | ---------- | ----------- | --------------- | ------------------- | ---- | ---- |  |
        |     | Cohort 1 |                  |          |                   |                  |                    |                  |                    |               |          |            |            |             |                 |                     |      |      |
        | New | street1  | 01/01/18         | 01/02/18 | Ready Plan 1 (RH) | Base             | 149                | 100              | 100                | 100           | 3        | 2          | 1          | Elevation 1 | Palette 1       | fpo1fpo2fpo3+1 more | aoo1 | iso1 |  |
        "
      `);
```

**Example from a project schedule table:**
```ts
    expect(tableSnapshot(r)).toMatchInlineSnapshot(`
      "
      |  | Task                | Start    | End      | Duration | Duration Variance | Predecessor | Successor | Status      | Assignee | Trade Partner | Resources | Phase                       | SubPhase                     | Files | Actions |
      |  | ------------------- | -------- | -------- | -------- | ----------------- | ----------- | --------- | ----------- | -------- | ------------- | --------- | --------------------------- | ---------------------------- | ----- | ------- |
      |  | Schedule Phase 1    | 01/01/18 | 01/10/18 | 8 days   |                   |             |           | 25%         |          |               |           |                             |                              |       |         |
      |  | Schedule SubPhase 1 | 01/01/18 | 01/02/18 | 1 day    |                   |             |           | 33%         |          |               |           |                             |                              |       |         |
      |  | Schedule SubPhase 2 | 01/01/18 | 01/02/18 | 1 day    |                   |             |           | 0%          |          |               |           |                             |                              |       |         |
      |  | 1Task 2             | 01/01/18 | 01/02/18 | 1 day    |                   | 1 +3 more   | 1 +4 more | IN_PROGRESS | User #2  | ABC Builders  | false     | Schedule PhaseGlobalPhase 1 | Schedule Sub Phase+ SubPhase |       | 2       |
      |  | Schedule Phase 2    | 01/01/18 | 01/02/18 | 1 day    |                   |             |           | 0%          |          |               |           |                             |                              |       |         |
      |  | Schedule SubPhase 3 | 01/01/18 | 01/02/18 | 1 day    |                   |             |           | 0%          |          |               |           |                             |                              |       |         |
      |  | Schedule SubPhase 4 | 01/01/18 | 01/02/18 | 1 day    |                   |             |           | 0%          |          |               |           |                             |                              |       |         |
      |  | Schedule Phase 3    |          |          |          |                   |             |           | 0%          |          |               |           |                             |                              |       |         |
      "
    `);
```

**Potential Improvements:**

- Config to ignore header row(s)
- Config to ignore columns
- Config to print text content differently for `input` types values (such as including labels)

### CODE REVIEW

File: src/utils/rtl.tsx [hunk 1]
```diff
@@ -89,6 +89,89 @@ export function rowAnd(r: RenderResult, rowNum: number, testId: string): HTMLEle
...
+
+  const headerDividerRowNumber = hasExpandableHeader ? 2 : 1;
+  rowsWithPaddingAndDividers.splice(headerDividerRowNumber, 0, wrapTableRowEnds(headerDivider));
+
+  // Pad a newline on top and bottom for cleaner diffs
+  return `\n${rowsWithPaddingAndDividers.join("\n")}\n`;
+}
+
+/** Prefer showing a `value` from a mocked input vs. the combined text content from an inputs markup */
+function getTextFromTableCellNode(node: ChildNode) {
```
Comment (stephenh) [inline src/utils/rtl.tsx:158]: Neat!

File: src/components/Table/GridTable.test.tsx [hunk 2]
```diff
@@ -2792,21 +2792,72 @@ describe("GridTable", () => {
...
+        "
+      `);
 
       //  When clicking to collapse `columnA`
       click(r.expandableColumn);
 
       // Then the column is collapsed
       expect(row(r, 1).childNodes).toHaveLength(1);
+      expect(tableSnapshot(r)).toMatchInlineSnapshot(`
+              "
```
Comment (stephenh) [inline src/components/Table/GridTable.test.tsx:2810]: Really minor question, but why do you think the `"` is indented here all the way under the `a` vs. under the `p` in `expect` like on line 2796? :shrug:

**Review by stephenh — APPROVED** [review]
Praise: this is amazing! Will be _so_ much better than our current assertions

File: src/components/Table/GridTable.test.tsx (hunk 2 above)
Comment (blambillotte) [reply src/components/Table/GridTable.test.tsx:2810]: ah I might have accidentally indented it when testing 🤔 
Testing a bit it looks like `toMatchInlineSnapshot` ignores the whitespace so the test passes either way. Generating from scratch does give the indentation we'd expect, will push that up.

**Pull request merged**